pkg/deploy/controller/deployment/controller.go

```diff
@@ -275,6 +275,8 @@ func (c *DeploymentController) makeDeployerPod(deployment *kapi.ReplicationContr
 	// Assigning to a variable since its address is required
 	maxDeploymentDurationSeconds := deployapi.MaxDeploymentDurationSeconds
 
+	gracePeriod := int64(10)
```

Comment:
Yes, the deployment controller is deleting all hooks and the deployer for the deployment it is reconciling:
https://github.com/openshift/origin/blob/26012f2fe41a6518f6ffd97c8b780c6d2c9c59e4/pkg/deploy/controller/deployment/controller.go#L365
We also test this in unit tests, eg:
https://github.com/openshift/origin/blob/26012f2fe41a6518f6ffd97c8b780c6d2c9c59e4/pkg/deploy/controller/deployment/controller_test.go#L466